common/src/chain/transaction/input.rs

```diff
@@ -22,12 +22,50 @@ impl From<Id<Block>> for OutPointSourceId {
     }
 }
 
-#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd, Encode, Decode)]
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
 pub struct OutPoint {
     id: OutPointSourceId,
     index: u32,
 }
 
+fn outpoint_source_id_as_monolithic_tuple(v: &OutPointSourceId) -> (u8, H256) {
+    let tx_out_index = 0;
+    let blk_reward_index = 1;
```

Comment:
Minor: these could be constants, i.e. `const TX_OUT_INDEX: u8 = 0;` etc.

Reply:
Even more minor: this could be a method `OutPointSourceId::as_monolithic_tuple(&self)`